app/src/org/odk/collect/android/utilities/GeoUtils.java

```diff
@@ -79,7 +80,11 @@ public static void showNoGpsDialog(Activity activity,
             factory.makeCancelable();
             factory.setOnCancelListener(onCancel);
         }
-        factory.showDialog();
+        if (activity instanceof CommCareActivity) {
+            ((CommCareActivity)activity).showAlertDialog(factory);
+        } else {
+            factory.showDialog();
```

Comment:
Add a note that in this case the dialog isn't persisted between orientation changes